webserver/templates/user/profile.html

```diff
@@ -3,53 +3,18 @@
 {% block title %}User "{{ user.musicbrainz_id }}" - ListenBrainz{% endblock %}
 
 {% block content %}
-  <h2 class="page-title">Most recent listens for {{ user.musicbrainz_id }}</h2>
-  <table id="listens" class="table table-striped table-bordered">
-    <tr><th>track</th><th>artist</th><th>time</th></tr>
-    <tr id="placeholder"><td>Loading your listens...</td></tr>
-  </table>
-
+  <div id="user-profile">
+    <h2 class="page-title">Most recent listens for {{ user.musicbrainz_id }}</h2>
+    <div id="listens-viewer-container" data-user-id="{{ user.musicbrainz_id }}">
+      <span class="loading-placeholder">Loading you listens...</span>
+    </div>
+  </div>
 {% endblock %}
-{% block scripts %}
-{{ super() }}
-<script src="{{ url_for('static', filename='js/lib/jquery.timeago.js') }}"></script>
-<script type="text/javascript">
-$(document).ready(function()
-{
-    jQuery("abbr.timeago").timeago();
-    load_listens();
-});
-
-function load_listens()
-{
-    $.ajax({
-            url: "/listen/user/{{ user.musicbrainz_id }}",
-            dataType : "json",
-            success: function(json)
-            {
-                load_table(json);
-            },
-            error: function(xhr, txtStatus, errorThrown)
-            {
-                console.log("Load data call failed");
-            }
-    });
-}
 
-function load_table(payload)
-{
-    $("#listens tr").remove();
-    $("#listens").append("<tr><th>artist</th><th>track</th><th>time</th></tr>");
-
-    listens = payload.payload.listens;
-    for(i = 0; i < listens.length; i++)
-    {
-        var s = new Date(listens[i]['listened_at'] * 1000).toISOString();
-        $('#listens').append("<tr><td>" + listens[i]['track_metadata']['artist_name'] + "</td>"
-                             + "<td>" + listens[i]['track_metadata']['track_name'] + "</td>" 
-                             + '<td><abbr class="timeago" title="' + s + '"></abbr></td></tr>') 
-    }
-    jQuery("abbr.timeago").timeago();
-}
-</script>
+{% block scripts %}
+  {{ super() }}
+  <script src="{{ url_for('static', filename='js/lib/react.js') }}"></script>
+  <script src="{{ url_for('static', filename='js/lib/JSXTransformer.js') }}"></script>
```

Comment:
Oh yeah, I still need to figure out how to build all the JSX/JavaScript. In AcousticBrainz too, which is where I copied that from.